src/main/java/com/cloudbees/jenkins/plugins/bitbucket/filesystem/BitbucketSCMFile.java

```diff
@@ -33,34 +33,36 @@
 
 public class BitbucketSCMFile  extends SCMFile {
 
-	private BitbucketApi api;
-	private String ref;
+	private final BitbucketApi api;
+	private final String ref;
+	private final String hash;
 	
 	public String getRef() {
 		return ref;
 	}
 
-	public void setRef(String ref) {
-		this.ref = ref;
-	}
-
 	public BitbucketSCMFile(BitbucketSCMFileSystem bitBucketSCMFileSystem,
 							BitbucketApi api,
-							String ref) {
+							String ref, String rev) {
 		super();
 		type(Type.DIRECTORY);
 		this.api = api;
 		this.ref = ref;
+		this.hash = rev;
 	}
 	
-    public BitbucketSCMFile(@NonNull BitbucketSCMFile parent, String name, Type type) {
+    public BitbucketSCMFile(@NonNull BitbucketSCMFile parent, String name, Type type, String rev) {
     	super(parent, name);
     	this.api = parent.api;
     	this.ref = parent.ref;
+    	this.hash = rev;
     	type(type);
-    	
     }
-    
+
+	public String getHash() {
+		return hash;
+	}
+
```

Comment:
Breaking binary compatibility by removing previous constructor signature

Reply:
Addressed all your concerns. Only exception is BitbucketRepositorySource. I think its fine keeping it with change in signature.